File: src/main/java/de/pflugradts/passbird/domain/model/namespace/Namespace.java
package de.pflugradts.passbird.domain.model.namespace;

import de.pflugradts.passbird.domain.model.transfer.Bytes;

public class Namespace {

    public static final Namespace DEFAULT = new Namespace(Bytes.bytesOf("Default"), NamespaceSlot.DEFAULT);

    private final Bytes bytes;
    private final NamespaceSlot slot;

    private Namespace(final Bytes bytes, final NamespaceSlot namespaceSlot) {
        this.bytes = bytes.copy();
        this.slot = namespaceSlot;
    }

    public Bytes getBytes() {
        return bytes;
    }

    public NamespaceSlot getSlot() {
        return slot;
    }

    static Namespace create(final Bytes bytes, final NamespaceSlot namespaceSlot) {
        return new Namespace(bytes, namespaceSlot);
    }

}

File: src/main/kotlin/de/pflugradts/passbird/domain/model/namespace/Namespace.kt
package de.pflugradts.passbird.domain.model.namespace

import de.pflugradts.passbird.domain.model.transfer.Bytes
import de.pflugradts.passbird.domain.model.transfer.Bytes.Companion.bytesOf

class Namespace private constructor(val bytes: Bytes, val slot: NamespaceSlot) {
    override fun toString() = "Namespace(slot=$slot)"
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Namespace
        return slot == other.slot
    }
    override fun hashCode() = slot.hashCode()
    companion object {
        val DEFAULT = Namespace(bytesOf("Default"), NamespaceSlot.DEFAULT)
        fun createNamespace(bytes: Bytes, namespaceSlot: NamespaceSlot) = Namespace(bytes.copy(), namespaceSlot)
    }
}
